pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -67,6 +67,15 @@ type EventListenerTrigger struct {
 	// +optional
 	Name         string              `json:"name,omitempty"`
 	Interceptors []*EventInterceptor `json:"interceptors,omitempty"`
+	// ServiceAccount optionally associates credentials with each trigger;
+	// more granular authorization for
+	// who is allowed to utilize the associated pipeline
+	// vs. defaulting to whatever permissions are associated
+	// with the entire EventListener and associated sink facilitates
+	// multi-tenant model based scenarios
+	// TODO do we want to restrict this to the event listener namespace and just ask for the service account name here?
```

Comment:
Yeah, having a cluster role that has access to get all serviceAccounts seems like a lot of permissions for a EventListener

Reply:
It could be role / role bindings for SAs in specific namespaces vs. a cluster role etc. but same basic notion.  It would be consistent with what is their for the service account and the event listener.

Looking back part of what gave me pause was the gitlab/github webhook interceptors had secret refs that allow the setting of namespaces.

```
// SecretRef contains the information required to reference a single secret string
// This is needed because the other secretRef types are not cross-namespace and do not
// actually contain the "SecretName" field, which allows us to access a single secret value.
type SecretRef struct {
	SecretKey  string `json:"secretKey,omitempty"`
	SecretName string `json:"secretName,omitempty"`
	Namespace  string `json:"namespace,omitempty"`
}

```

I don't quite follow the comment, and have not discovered through code exploration why cross namespace is needed with those.

Can you clarify @dibyom ?

Reply:
Yeah, I think I've only used secrets in the same namespace as well...Maybe we can get rid of the `Namespace` from those refs as well

Reply:
Isn't this use when Trigger is in different namespace? Then secret is generally also in the Trigger Namespace. Otherwise, EL ns is used.

Reply:
https://github.com/tektoncd/triggers/blob/master/pkg/interceptors/interceptors.go#L33

Reply:
Yep thanks for the pointer @khrm 

And the next step for that is to do the signature validation so that the secret contents in the payload from github/gitlab that have arrived through the event sink match up ...i.e. 
https://github.com/tektoncd/triggers/blob/master/pkg/interceptors/github/github.go#L71

So ... characterizing "current support"
- we allow that secret to reside in a separate namespace from the `EventListener` and `EventListenerTrigger` namespace.
- meaning the `EventListener` SA today needs to be able to read `Secret` objects from namespaces other than its own.

What is unclear to me ... and I see no explanation of this in https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md , is what scenario is served by letting this secret reside in namespaces besides the `EventListener` / `EventListenerTrigger`.  

Is it that the pipelines/tasks/pipelineruns/taskruns referenced by the trigger may be in the namesapce this secret resides in, and we are assuming the `EventListener` SA has permission to create those object in those namespaces?  And because of that, it makes sense for the owner of those to create the github/gitlab secret in the same namespace as the pipelines/task/pipelineruns/taskruns ?

If so, the doc really needs to clarify that.

If it is something else, please explain.

Now, on to this PR, and what we do with it, along with your feature https://github.com/tektoncd/triggers/issues/456 

I think I've come to this understanding.  That feature, and this PR, are around creating/access pipeline/tasks/pipelineruns/taskruns in a namespace that differs from the `EventListener` namespace **WITHOUT** adding permissions to create/access pipeline/tasks/pipelineruns/taskruns in that namesapce to the `EventListener` SA (in addition to being able to create those in the event listener's namespace).

Is that correct?  If so, yes this feature facilitates that.  If that is not correct, please clarify.

But per our discussion, the current work around is to create different `EventListener` instances in each namespace.  Where the discussion today speculated that isn't "too expensive".

Can you elaborate beyond what I mentioned today why that work around would be untenable for you?

thanks

Reply:
We are thinking of having only a single deployment for EventListener to reduce number of pods related to EventListener in a cluster. So that's why we give cluster role permission to EL to act as multitenant EL.

Reply:
OK so the motivation is the concern over the cost of EventListeners, which was touch upon during the last sync call.  Perhaps we can revisit it again on today's call.

Reply:
Yes, this seems worth a discussion in today's call! :+1: